biojava-structure/src/main/java/org/biojava/nbio/structure/basepairs/BasePairParameters.java

```diff
@@ -0,0 +1,718 @@
+
+        cosgamma = A[2][2];
+        if (cosgamma > 1.0) cosgamma = 1.0;
+        else if (cosgamma < -1.0) cosgamma = -1.0;
+
+        gamma = acos(cosgamma);
+
+        sgcp = A[1][1]*A[0][2]-A[0][1]*A[1][2];
+
+        if (gamma == 0.0) omega = -atan2(A[0][1],A[1][1]);
+        else omega = atan2(A[2][1]*A[0][2]+sgcp*A[1][2],sgcp*A[0][2]-A[2][1]*A[1][2]);
+
+        omega2_minus_phi = atan2(A[1][2],A[0][2]);
+
+        phi = omega/2.0 - omega2_minus_phi;
+
+        M[0] = gamma*sin(phi)*180.0/PI;
+        M[1] = gamma*cos(phi)*180.0/PI;
+        M[2] = omega*180.0/PI;
+
+        sm = sin(omega/2.0-phi);
+        cm = cos(omega/2.0-phi);
+        sp = sin(phi);
+        cp = cos(phi);
+        sg = sin(gamma/2.0);
+        cg = cos(gamma/2.0);
+
+        M[3] = (cm*cg*cp-sm*sp)*A[0][3]+(sm*cg*cp+cm*sp)*A[1][3]-sg*cp*A[2][3];
+        M[4] = (-cm*cg*sp-sm*cp)*A[0][3]+(-sm*cg*sp+cm*cp)*A[1][3]+sg*sp*A[2][3];
+        M[5] = (cm*sg)*A[0][3]+(sm*sg)*A[1][3]+cg*A[2][3];
+
+        return M;
+
+    }
+
+
+    public static char complementBase(char base, boolean RNA) {
+        if (base == 'A' && RNA) return 'U';
+        if (base == 'A') return 'T';
+        if (base == 'T' && !RNA) return 'A';
+        if (base == 'U' && RNA) return 'A';
+        if (base == 'C') return 'G';
+        if (base == 'G') return 'C';
+        return ' ';
+    }
+
+    public static String complement(String sequence, boolean RNA) {
+        String result = "";
+        for (int i = sequence.length() - 1; i >= 0; i--) {
+            result += complementBase(sequence.charAt(i), RNA);
+        }
+        return result;
+    }
+
+    public static double[] cross(double[] a, double[] b) {
+        assert a.length >= 3 && b.length >= 3;
+        double[] result = new double[4];
+        result[0] = a[1]*b[2]-a[2]*b[1];
+        result[1] = a[2]*b[0]-a[0]*b[2];
+        result[2] = a[0]*b[1]-a[1]*b[0];
+        return result;
+    }
+
+    public static double[] removeComponent(double[] a, double[] b) {
+        double dot = 0;
+        double[] result = new double[4];
+        for (int i = 0; i < 3; i++) {
+            dot += a[i]*b[i];
+        }
+        for (int i = 0; i < 3; i++) {
+            result[i] = a[i]-dot*b[i];
+        }
+        return result;
+
+    }
+
+    public static String longestCommonSubstring(String s1, String s2) {
+        int start = 0;
+        int max = 0;
+        for (int i = 0; i < s1.length(); i++) {
+            for (int j = 0; j < s2.length(); j++) {
+                int x = 0;
+                while (s1.charAt(i + x) == s2.charAt(j + x)) {
+                    x++;
+                    if (((i + x) >= s1.length()) || ((j + x) >= s2.length())) break;
+                }
+                if (x > max) {
+                    max = x;
+                    start = i;
+                }
+            }
+        }
+        return s1.substring(start, (start + max));
+    }
+
+    protected static boolean match(char a, char b, boolean RNA) {
```

Comment:
If you don't know, I don't know!   I'll JavaDoc everything and fix the styling issues.